### PR TITLE
Make Rustfmt + Editorconfig consistent with parity-common

### LINKED ISSUE #117: Sync .rustfmt.toml with parity-common
Enforce a "tools team" rustfmt.

Perhaps the same as `jsonrpsee` (which comes from `parity-common`):

```toml
hard_tabs = true
max_width = 120
use_small_heuristics = "Max"
edition = "2018"
```

_Originally posted by @dvdplm in https://github.com/paritytech/substrate-archive/pull/108#discussion_r511873571_

### PR DESCRIPTION
**Pull request by insipx**
Should be merged after #125 #126 and #127 so that `rustfmt`may be run again on those PR's

fixes #117

### CODE REVIEW

File: .rustfmt.toml [hunk 1]
```diff
@@ -1,3 +1,4 @@
-newline_style = "Unix"
```
Comment (dvdplm) [inline .rustfmt.toml:1]: I don't think this will be a problem ever but maybe we should keep this one?

**Review by dvdplm — APPROVED** [review]
Didn't read through the code but I trust this is essentially the updated `.rustfmt` + `cargo fmt`?

**Review by niklasad1 — APPROVED** [review]

**Review by emostov — COMMENTED** [review]
Looks good! V excited about unified styling :) (assuming rustfmt was run after adding back Unix newline)

**Review by mattrutherford — APPROVED** [review]

**Comment by dvdplm** [comment]
> V excited about unified styling :)

I am too, which is absolutely ridiculous of course, but hey, happiness in the small things yes? :)

**Pull request merged**